truenas/webui
# NAS-129686 / 24.10 / Fix SMB ACL form

**Summary**

1. when AD cache is empty (see ticket), when **Group** combobox does not contain any suitable values to choose from,
I've changed to code to add the ability to type in a custom value from keyboard . When user presses **Save**, every value is validated for being a valid existing entry in the Active Directory 

2. Middleware returns special type of AD entry `BOTH` , and when AD cache is empty (see ticket) it's not possible to distinguish a user from a group. 
   It was agreed that, it would be unreasonable to try to squeeze in last-second middleware update for 24.04.2.
   So, in such cases, the WebUI code was updated to make an attempt of detecting the type, otherwise it treats an AD entry as `GROUP`.

**Testing**

1. Log into WebUI
2. Setup the Active directory
4. Disable AD User / Group Cache and rebuild cache
5. Save the configuration
6. Create an SMB share
7. Edit the SMB ACL
8. Add a known to exist AD Group using keyboard entry, not picking any of options of the dropdown menu (for example `AD02\domain users`)
  and/or add a User entry by typing on the keyboard (for example type in `AD02\administrator`)
9. Save the ACL
10. Edit the SMB ACL
11. Observe the AD Group and/or User entry
